src/spc/codegen.jl

```diff
@@ -0,0 +1,486 @@
+using StaticArrays
+
+struct CodegenState
+    sema::SemaResult
+end
+
+LString(s::SNode{<:SP.Terminal}) = lowercase(String(s))
+LString(s::SNode{<:SP.AbstractASTNode}) = lowercase(String(s))
+LString(s::SNode{<:SC.Terminal}) = String(s)
+LString(s::SNode{<:SC.AbstractASTNode}) = String(s)
+LString(s::AbstractString) = lowercase(s)
+LString(s::Symbol) = lowercase(String(s))
+LSymbol(s) = Symbol(LString(s))
+
+function Base.LineNumberNode(n::SNode)
+    sf = n.ps.srcfile
+    lsf = sf.lineinfo
+    lno_first = SpectreNetlistParser.LineNumbers.compute_line(lsf, n.startof+n.expr.off)
+    LineNumberNode(lno_first, Symbol(sf.path))
+end
+
+function hasparam(params, name)
+    for p in params
+        if LString(p.name) == name
+            return true
+        end
+    end
+    return false
+end
+
+
+function cg_modelref!(state::CodegenState, modelref)
+
```

Comment:
Is this supposed to be empty?

Reply:
I moved model resolution to sema and the remaining caller is wrong, but that whole code path has bigger problems (https://github.com/JuliaComputing/CedarSim.jl/issues/853) that will need handling at some point (probably in sema).